### DISCUSSION TITLE
Easily transfer existing sources into git?

### DISCUSSION THREAD

**Discussion by LGSDEVS** [codefori/vscode-ibmi#682] [IBM i]
Hello all,

first of all I don't know much about Git, please forgive me if I ask stupid questions.

I have on the AS400 many libraries with many source files, these I would like to lift most simply into the Git. For this I watched the Youtube video (https://youtu.be/XuiGyWptgDA), unfortunately I did not understand it completely and wanted to ask the questions here.

Here's what I plan to do:

I have a libraries TESTLIB in this are about 200 source files (DDS, CL, RPGLE) these I have opened in the object browser, now I would have to create a similar structure manually in IFS according to the Youtube video.

1. Why do I need the IFS directory?
2. Do I really have to manually transfer all my 200 sources to the IFS directory or is there an automatism for this?
3. Can I tell the object browser to download not only individual source files but directly the structure with the libraries, files and source files? So I could quickly make a copy on my local PC and upload this to the Git if necessary.

**Comment by andreas-prouza** [comment]
1. It's more comfortable to stay in the IFS. The RDi also supports working with GIT and source files in a library, but it has many disadvantages. (E.g.: you need to use RDi and can't easily use other IDEs, also if you compare versions it's better not to have the change timestamp and the sequence number for each row in the source)
2. You can write a small RPG to read (via SQL) all members and make a CPYTOSTMF to do this automatically
3. see 2.

**Reply by worksofliam** [reply]
To point number two, see the migrate tool! Does something just like your description. https://github.com/worksofliam/migrate
